### PR TITLE
Added BlueROV2 mesh for visualization

### PR DESCRIPTION
**Pull request by evan-palmer**
gazebo

## Changes Made

This PR updates the BlueROV2 visualization to use a mesh generated from the [BlueROV2 Solidworks model](https://grabcad.com/library/bluerov2-1). Additionally, T200 propeller meshes obtained from the [T200 Solidworks model](https://grabcad.com/library/bluerobotics-t200-thruster-1) have been included to support propeller visualization.

## Files Changed

- `/models/bluerov2/meshes/**`: Introduced three new meshes: one for the BlueROV2 with the thrusters angled at 45 degrees and two for the T200 propellers (including CW & CCW)
- `/models/bluerov2/model.sdf.in`: Updated the visual elements to use the added meshes
- `/scripts/generate_model.py`: Adjusted the placement of the thrusters in the model
- `/images/gazebo.png`: Modified the image to use the updated meshes

### CODE REVIEW

**Comment by evan-palmer** [comment]
I'd like to note, that I am in the process of switching over the STL files to COLLADA so that the resulting models preserve their coloring. I will update the PR once they have been updated.

**Comment by clydemcqueen** [comment]
@evan-palmer Thank you for contributing! This is a great addition!

It would be great to preserve the color if possible.

If it's easy, perhaps you could simplify the files a bit by removing the holes in the penetrators and other little bits that don't render well.

When this is ready, perhaps @srmainwaring will be interested in adding the model to https://github.com/ArduPilot/ardupilot_gazebo.

File: models/bluerov2/model.sdf [hunk 1]
```diff
@@ -21,28 +21,26 @@
...
       </inertial>
 
-      <visual name="base_link_box">
-        <pose>0 0 0 0 0 0</pose>
-        <transparency>0.5</transparency>
+      <visual name="base_link_visual">
+        <pose>-0.134 -0.4935 0.17 1.5707963267948966 0 1.5707963267948966</pose>
+        <geometry>
+          <mesh>
+            <uri>model://bluerov2/meshes/bluerov2-45.stl</uri>
```
Comment (clydemcqueen) [inline models/bluerov2/model.sdf:28]: Looks like you need to re-run generate_model again.

File: models/bluerov2/model.sdf.in [hunk 2]
```diff
@@ -21,25 +21,18 @@
         </inertia>
       </inertial>
 
-      <visual name="base_link_box">
-        <pose>0 0 0 0 0 0</pose>
-        <transparency>0.5</transparency>
+      <visual name="base_link_visual">
+        <pose>-0.134 -0.4935 0.17 @d90 0 @d90</pose>
```
Comment (clydemcqueen) [inline models/bluerov2/model.sdf.in:25]: Why isn't this (0, 0, 0)?

**Comment by srmainwaring** [comment]
> When this is ready, perhaps @srmainwaring will be interested in adding the model to https://github.com/ArduPilot/ardupilot_gazebo

@clydemcqueen sure! We'd add to https://github.com/ArduPilot/SITL_Models/tree/master/Gazebo which is where we've been placing models recently. I'll keep an eye out for when this PR is merged.

File: models/bluerov2/model.sdf (hunk 1 above)
Comment (evan-palmer) [reply models/bluerov2/model.sdf]: Sure enough :smile:

File: models/bluerov2/model.sdf.in (hunk 2 above)
Comment (evan-palmer) [reply models/bluerov2/model.sdf.in]: Nice catch! I didn't adjust the origin of the CAD file when generating the STL. I centered the geometry to (0, 0, 0) with the updated COLLADA files.

**Comment by evan-palmer** [comment]
> It would be great to preserve the color if possible.
> 
> If it's easy, perhaps you could simplify the files a bit by removing the holes in the penetrators and other little bits that don't render well.

Thank you for the review! With the latest commit, I switched over the STL files to COLLADA files to preserve the original coloring of the BlueROV2. I did change the coloring of the transparent parts (electronics enclosure, battery bottle, and camera dome) to light grey because those were pretty difficult to see in Gazebo. I spent a little while trying to resolve your comment regarding simplifying the penetrators and some of the other bits as well. I was able to improve the rendering quality with the new file format, but wasn't able to fill those penetrator holes up. The original CAD file is a Solidworks assembly which doesn't transfer super well to Inventor (what I have access too) and makes it a bit difficult to make changes to the bodies themselves. I also wanted to note that I wasn't able to get the back two propellers fully aligned without throwing off the top and front propeller alignment. I opted to leave those a little out of whack to preserve the accuracy of the more visible thrusters.

**Review by clydemcqueen — APPROVED** [review]
This is a huge improvement. Thanks a bunch!

**Pull request merged**